pkg/providers/cloudstack/cloudstack_test.go

```diff
@@ -247,6 +247,45 @@ func TestProviderGenerateCAPISpecForCreate(t *testing.T) {
 	test.AssertContentToFile(t, string(md), "testdata/expected_results_main_md.yaml")
 }
 
+func TestProviderGenerateCAPISpecForCreateWithAutoscalingConfiguration(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	setupContext(t)
+	ctx := context.Background()
+	kubectl := mocks.NewMockProviderKubectlClient(mockCtrl)
+	cluster := &types.Cluster{
+		Name: "test",
+	}
+	clusterSpec := givenClusterSpec(t, testClusterConfigMainFilename)
+
+	datacenterConfig := givenDatacenterConfig(t, testClusterConfigMainFilename)
+	machineConfigs := givenMachineConfigs(t, testClusterConfigMainFilename)
+
+	fmt.Println(len(clusterSpec.Cluster.Spec.WorkerNodeGroupConfigurations))
```

Comment:
Let's avoid using fmt.Println. If you're looking to debug a problem, you can use the GoLand or VSCode debugger instead to look at values in the stack

Reply:
Yup, missed this. Cleaning up.